modules/azurerm/Azure-LB-Rule/variables.tf

```diff
@@ -9,39 +9,20 @@
 #
 # --------------------------------------------------------------------------------------
 
-variable "name" {
-  description = "Specifies the name of the LB Rule. Changing this forces a new resource to be created."
-  type        = string
-}
-
 variable "loadbalancer_id" {
   description = "The ID of the Load Balancer in which to create the Rule. Changing this forces a new resource to be created."
   type        = string
 }
 
-variable "frontend_ip_configuration_name" {
-  description = "The name of the frontend IP configuration to which the rule is associated."
-  type        = string
-}
-
-variable "protocol" {
-  description = "The transport protocol for the external endpoint. Possible values are Tcp, Udp or All."
-  type        = string
-  default     = "All"
-}
-
-variable "frontend_port" {
-  description = "The port for the external endpoint. Port numbers for each Rule must be unique within the Load Balancer. Possible values range between 0 and 65534, inclusive."
-  type        = number
-}
-
-variable "backend_port" {
-  description = "The port used for internal connections on the endpoint. Possible values range between 0 and 65535, inclusive."
-  type        = number
-}
-
-variable "backend_address_pool_ids" {
-  description = "A list of reference to a Backend Address Pool over which this Load Balancing Rule operates."
-  type        = list(string)
+variable "rule_configuration" {
   default     = []
+  description = "The list of Azure LB rules"
```

Comment:
Let's use a map instead of a list of objects https://github.com/wso2/azure-terraform-modules/blob/184e905a3ae4ba355115f49edd6eb015aef65516/modules/azurerm/Firewall-LB-DNAT-Rule-Collection/firewall_dnat_rules.tf#L20